test/testdata/lsp/code_actions/move_method/instance_method.A.rbedited

```diff
@@ -0,0 +1,23 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module IncModule
+  extend T::Sig
+  sig {params(x: Integer).returns(Integer)}
+  def inc(x)
+    # ^ apply-code-action: [A] Move method to a new module
+    x + 1
+  end
+end
+
+class Inc
+  extend T::Sig
+
+  
+  
+end
+
+foo = Inc.new
+res = IncModule.inc(42)
```

Comment:
Because of cases like this, I think we should detect if the method the user has requested to refactor is an instance method, and not generate a refactor.extract code action in that case. My reasoning:

- The step here to rename the receiver is wrong. The code generated by the action fails to type check.
- If we attempt to make this code type check by changing `def inc` to be a singleton class method (`def self.inc`), that doesn't necessarily make sense.

I think it's fine to say that there's no code action here. What do you think?

Reply:
>The code generated by the action fails to type check.

That was a conscious design decision. The refactoring doesn't try to generate the correct code, but tries to automate the toil of moving the method out and updating the call sites. The result may be not correct, but the typecheck errors will guide the user on how to fix them.

>I think it's fine to say that there's no code action here.

I also think it's fine too. However, I can still see the value of leaving it. The refactoring doesn't generate the correct code, but the type errors, will help you to fix them. Does this refactoring make sense here is the question to the users and what their goals are. If the user is aware of all of that, the code action can still automate some toil for them.

But I can agree with you here, that behavior is a bit surprising, and if we want to follow the principle of least surprise, it'd make sense to limit the refactoring to class methods only.

Reply:
yeah in the past people have been more annoyed with autocorrects that are wrong than with those that don't exist. Especially in cases where it's possible to quickly check to know that some autocorrect will be wrong, I like to err on the side of not doing it.